packages/pigeon/test/pigeon_lib_test.dart

```diff
@@ -416,4 +416,26 @@ void main() {
     expect(parseResult.errors[0].message, contains('Generic fields'));
     expect(parseResult.errors[0].lineNumber, isNotNull);
   });
+
+  test('error with static field', () {
+    final Pigeon dartle = Pigeon.setup();
+    const String code = '''
+class WithStaticField {
+  static int? x;
+  int? y;
+}
+
+@HostApi()
+abstract class WithStaticFieldApi {
+  void doit(WithStaticField withTemplate);
+}
+''';
+    _withTempFile('compilationError.dart', (File file) {
```

Comment:
What purpose is making the name of the file a parameter serving? Are there cases where the name of the test file matters?